java/src/main/java/io/vertx/ext/eventbus/client/AsyncResult.java

```diff
@@ -0,0 +1,31 @@
+package io.vertx.ext.eventbus.client;
+
+/**
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class AsyncResult<T> {
+
+  private final T result;
+  private final Throwable cause;
+
+  public AsyncResult(T result, Throwable cause) {
```

Comment:
make it private and make two static method to build it : success(T) and failure(Throwable)

Reply:
Static factory methods for generic types look a bit clunky - I hope this is what you intended it to be :)